File: my-club-web/src/model/section.ts
import {Team} from "./team";

export class Section {
  constructor() {
    this.name = ""
    this.description = ""
  }

  id: number
  clubId: number

  name: string
  description: string

  teams: Team[]
}

export const sectionsPath: string = "/api/section/";

File: my-club-api/src/main/java/com/my/club/model/Section.java
package com.my.club.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import io.quarkus.hibernate.orm.panache.PanacheQuery;
import jakarta.persistence.*;

import java.util.List;

@Entity
public class Section extends PanacheEntity {

    public String name;

    public String description;

    @ManyToOne
    @JoinColumn(name = "club_id")
    public Club club;

    @OneToMany(targetEntity = Team.class, cascade = CascadeType.ALL, orphanRemoval = true)
    public List<Team> teams;

    @ManyToMany
    List<Member> members;

    public static List<Section> findByClubId(long clubId) {
        PanacheQuery<Section> clubSections = find("club.id=?1", clubId);
        return clubSections.list();
    }

}
